openpype/hosts/traypublisher/plugins/create/create_online.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+"""Creator of online files.
+
+Online file retain their original name and use it as subset name. To
+avoid conflicts, this creator checks if subset with this name already
+exists under selected asset.
+"""
+from pathlib import Path
+
+from openpype.client import get_subset_by_name, get_asset_by_name
+from openpype.lib.attribute_definitions import FileDef
+from openpype.pipeline import (
+    CreatedInstance,
+    CreatorError
+)
+from openpype.hosts.traypublisher.api.plugin import TrayPublishCreator
+
+
+class OnlineCreator(TrayPublishCreator):
+    """Creates instance from file and retains its original name."""
+
+    identifier = "io.openpype.creators.traypublisher.online"
+    label = "Online"
+    family = "online"
+    description = "Publish file retaining its original file name"
+    extensions = [".mov", ".mp4", ".mxf", ".m4v", ".mpg"]
+
+    def get_detail_description(self):
+        return """# Create file retaining its original file name.
+
+        This will publish files using template helping to retain original
+        file name and that file name is used as subset name.
+
+        Bz default it tries to guard against multiple publishes of the same
+        file."""
+
+    def get_icon(self):
+        return "fa.file"
+
+    def create(self, subset_name, instance_data, pre_create_data):
+        if not pre_create_data.get("representation_file")["filenames"]:
+            raise CreatorError("No files specified")
+
+        asset = get_asset_by_name(self.project_name, instance_data["asset"])
+        origin_basename = Path(pre_create_data.get(
+            "representation_file")["filenames"][0]).stem
+
+        if get_subset_by_name(
+                self.project_name, origin_basename, asset["_id"]):
+            raise CreatorError(f"subset with {origin_basename} already "
+                               "exists in selected asset")
+
+        instance_data["originalBasename"] = origin_basename
+        subset_name = origin_basename
+        path = (Path(pre_create_data.get("representation_file")["directory"]) / pre_create_data.get("representation_file")["filenames"][0]).as_posix()  # noqa
+
+        instance_data["creator_attributes"] = {"path": path}
+
+        # Create new instance
+        new_instance = CreatedInstance(self.family, subset_name,
+                                       instance_data, self)
+        self._store_new_instance(new_instance)
```

Comment:
I'm sorry but "What a mess...". `pre_create_data.get("representation_file")` called 4 times without validated output. Key `"originalBasename"` stored to instance data but is not used. Too long line...

Quick suggestion:
```suggestion
    def create(self, subset_name, instance_data, pre_create_data):
        representation_file = pre_create_data.get("representation_file")
        if not representation_file:
            raise CreatorError("Representation files information is missing")

        filenames = representation_file["filenames"]
        if not filenames:
            raise CreatorError(
                "No files specified (You didn't drop any files)")

        filename = Path(filenames[0])
        origin_basename = filename.stem

        # Validate if subset already exists
        asset_name = instance_data["asset"]
        asset_doc = get_asset_by_name(
            self.project_name, asset_name, fields=["_id"])
        if not asset_doc:
            raise CreatorError(f"Didn't fint asset with name \"{asset_name}\"")

        subset_doc = get_subset_by_name(
            self.project_name, origin_basename, asset["_id"], fields=["_id"])
        if subset_doc:
            raise CreatorError(f"subset with {origin_basename} already "
                               "exists in selected asset")

        path = Path(representation_file["directory"]) / filename

        instance_data["creator_attributes"] = {"path": path.as_posix()}

        # Create new instance
        new_instance = CreatedInstance(
            self.family, origin_basename, instance_data, self)
        self._store_new_instance(new_instance)
```